test/feature/middleware_support.feature

```diff
@@ -0,0 +1,153 @@
+Feature: Middleware support
+
+  Scenario Outline: simple middleware support
+    Given a file "routes.coffee" with the content
+      """
+      module.exports = ({ GET }) ->
+        GET '/users', to: 'users#index'
+      """
+    And a file "controllers/users_controller.coffee" with the content
+      """
+      middle = (req, res, next) ->
+        req.custom = 'foo'
+        next()
+
+      class UsersController
+        middleware:
+          index: middle
+
+        index:   (req, res) -> res.end req.custom
```

Comment:
Yeah, `useMiddleware` is nice since that's what Express calls it, but I've been trying to follow rails conventions in the routes file as much as possible, and using some rails conventions in controllers might make sense too. I think either would be fine as long as it's well documented.